sp/src/game/server/hl2/hl2_player.cpp

```diff
@@ -4837,6 +4839,50 @@ void CHL2_Player::TraceKickAttack( CBaseEntity* pKickedEntity )
 	}
 }
 
+bool CHL2_Player::TryRagdollKickedEnemy(CBaseEntity* pKickedEntity, trace_t* tr, CTakeDamageInfo* dmgInfo, CBaseEntity* pKickingEntity)
+{
+	if (!sv_player_kick_attack_ragdolls.GetBool())
+	{
+		return false;
+	}
+
+	if (pKickedEntity->MyNPCPointer() == NULL)
+	{
+		return false;
+	}
+
+	// If the victim does not have a humanoid or small hull, short circuit
+	if (pKickedEntity->MyNPCPointer()->GetHullType() >= HULL_WIDE_SHORT)
+	{
+		return false;
+	}
+
+	if (!pKickedEntity->MyNPCPointer()->CanBecomeRagdoll() || !pKickedEntity->MyNPCPointer()->CanBecomeServerRagdoll())
+	{
+		return false;
+	}
+
+	// If the victim has more than 100 HP, short circuit
+	if (GetHealth() > 100)
+	{
+		return false;
+	}
+
+	CRagdollMagnet* pMagnet = CRagdollMagnet::FindBestMagnet(pKickedEntity);
+	if (pMagnet)
+	{
+		DevMsg("Found ragdoll magnet for kicked NPC!\n");
+		// Send the damage to the recipient
+		Vector vecAim = BaseClass::GetAutoaimVector(AUTOAIM_SCALE_DEFAULT);
```

Comment:
We need to check if the aiming vector is within a certain angle of the magnet direction